### PR TITLE
Update API docs for `required_fields`

### PR DESCRIPTION
**Pull request by nikitaindik**
**Epic: https://github.com/elastic/security-team/issues/1974**
**Related to: https://github.com/elastic/kibana/issues/173594**

## Summary

API updates for https://github.com/elastic/kibana/pull/180682.

Updates related API docs with `required_field` changes made in https://github.com/elastic/kibana/pull/180682.

### CODE REVIEW

File: docs/detections/api/rules/rules-api-create.asciidoc [hunk 1]
```diff
@@ -342,6 +342,14 @@ single execution. Defaults to `100`.
 |references |String[] |Array containing notes about or references to
 relevant information about the rule. Defaults to an empty array.
 
+|required_fields |Object[] a| Elasticsearch fields and their types that need to be present for the rule to function. The object has these fields:
+
+ * `name` (string, required): Field name
+ * `type` (string, required): Field type
+ * `ecs` (boolean, optional): Indicates whether the field is {ecs-ref}[ECS-compliant]. This property is only present in responses. Its value is computed based on field's name and type.
+
+*NOTE*: The value of `required_fields` does not affect the rule's behavior. It can be thought of as a way to document the fields that the rule expects to be present in the data. Specifying `required_fields` incorrectly won't cause the rule to fail.
```
Comment (joepeeples) [inline docs/detections/api/rules/rules-api-create.asciidoc:351]:
```suggestion
|required_fields |Object[] a| {es} fields and their types that need to be present for the rule to function. The object has these fields:

 * `name` (string, required): The field's name.
 * `type` (string, required): The field's data type.
 * `ecs` (Boolean, optional): Indicates whether the field is {ecs-ref}[ECS-compliant]. This property is only present in responses. Its value is computed based on field's name and type.

*NOTE*: The value of `required_fields` does not affect the rule's behavior, and specifying it incorrectly won't cause the rule to fail. Use `required_fields` as an informational property to document the fields that the rule expects to be present in the data. 
```

File: docs/detections/api/rules/rules-api-update.asciidoc [hunk 2]
```diff
@@ -244,6 +244,14 @@ single execution. Defaults to `100`.
 |references |String[] |Array containing notes about or references to
 relevant information about the rule. Defaults to an empty array.
 
+|required_fields |Object[] a| Elasticsearch fields and their types that need to be present for the rule to function. The object has these fields:
+
+ * `name` (string, required): Field name
+ * `type` (string, required): Field type
+ * `ecs` (boolean, optional): Indicates whether the field is {ecs-ref}[ECS-compliant]. This property is only present in responses. Its value is computed based on field's name and type.
+
+*NOTE*: The value of `required_fields` does not affect the rule's behavior. It can be thought of as a way to document the fields that the rule expects to be present in the data. Specifying `required_fields` incorrectly won't cause the rule to fail.
```
Comment (joepeeples) [inline docs/detections/api/rules/rules-api-update.asciidoc:253]:
```suggestion
|required_fields |Object[] a| {es} fields and their types that need to be present for the rule to function. The object has these fields:

 * `name` (string, required): The field's name.
 * `type` (string, required): The field's data type.
 * `ecs` (Boolean, optional): Indicates whether the field is {ecs-ref}[ECS-compliant]. This property is only present in responses. Its value is computed based on field's name and type.

*NOTE*: The value of `required_fields` does not affect the rule's behavior, and specifying it incorrectly won't cause the rule to fail. Use `required_fields` as an informational property to document the fields that the rule expects to be present in the data. 
```

**Review by joepeeples — COMMENTED** [review]
Thanks for preparing this PR! I added a few possible edits for consideration.

**Comment by nikitaindik** [comment]
Thanks @joepeeples! I have committed your suggestions.

**Review by joepeeples — APPROVED** [review]

**Review by dplumlee — APPROVED** [review]
LGTM!

**Pull request merged**